sway-lsp/src/core/token.rs

```diff
@@ -247,15 +281,8 @@ pub fn type_info_to_symbol_kind(
 pub fn get_range_from_span(span: &Span) -> Range {
     let start = span.start_pos().line_col();
     let end = span.end_pos().line_col();
-
-    let start_line = start.0 as u32 - 1;
-    let start_character = start.1 as u32 - 1;
-
-    let end_line = end.0 as u32 - 1;
-    let end_character = end.1 as u32 - 1;
-
     Range {
-        start: Position::new(start_line, start_character),
-        end: Position::new(end_line, end_character),
+        start: Position::new(start.0 as u32 - 1, start.1 as u32),
```

Comment:
Just to be sure I understand, we are shifting the character by 1 to be 0-indexed in line with the [LSP spec](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#range)?

Reply:
Yep that's correct. We've been doing this in this function before hand, i've removed the variable allocations now.